source/common/stats/thread_local_store.h

```diff
@@ -262,6 +302,19 @@ class ThreadLocalStoreImpl : Logger::Loggable<Logger::Id::stats>, public StoreRo
                            StatMap<std::shared_ptr<StatType>>* tls_cache,
                            StatNameHashSet* tls_rejected_stats, StatType& null_stat);
 
+    /**
+     * Looks up an existing stat, populating the local cache if necessary. Does
+     * not check the TLS or rejects, and does not create a stat if it does not
+     * exist.
+     *
+     * @param name the full name of the stat (not tag extracted).
+     * @param central_cache_map a map from name to the desired object in the central cache.
+     * @return a reference to the stat, if it exists.
+     */
+    template <class StatType>
+    absl::optional<std::reference_wrapper<const StatType>>
+    findStat(StatName name, StatMap<std::shared_ptr<StatType>>& central_cache_map) const;
```

Comment:
one other thought (not required for resolution in this PR): ideally you might want to give an EXCLUSIVE_LOCKS_REQUIRED annotation here, but I think maybe the reason lock annotations are not used in this class because the compiler's analysis gets confused with parent_.mutex_ .

But one pattern I've been doing -- not sure where I first picked it up -- is to suffix functions that are expected to be called under lock with "LockHeld", e.g. findStatLockHeld().

Up to you.

Reply:
+1 that would be nice if you don't mind doing it. (Or figuring out a way to fix the mutex annotations entirely)

Reply:
Renamed to findStatLockHeld. I may try to fix the mutex annotations some other time--agreed that would be ideal.